cleanup_snapshots.tf

```diff
@@ -0,0 +1,121 @@
+data "archive_file" "cleanup_snapshots_zip" {
+  type        = "zip"
+  source_file = "${local.path_module}/lambda/code/cleanup_snapshots.py"
+  output_path = "${local.path_module}/lambda/package/cleanup_snapshots.zip"
+}
+
+resource "aws_lambda_function" "cleanup_snapshots" {
+  filename         = "${path.module}/lambda/cleanup_snapshots/package/lambda.zip"
+  function_name    = "${var.pipeline_name}-${var.namespace}-cleanup-snapshots"
+  role             = "${aws_iam_role.cleanup_snapshots.arn}"
+  handler          = "cleanup_snapshots.lambda_handler"
+  source_code_hash = "${data.archive_file.cleanup_snapshots_zip.output_base64sha256}"
+  runtime          = "python3.7"
+  timeout          = "900"
+  memory_size      = "128"
+
+  vpc_config {
+    subnet_ids         = ["${var.lambda_subnet}", "${var.lambda_subnet_az2}"]
+    security_group_ids = ["${var.lambda_sgrp}"]
+  }
+
+  tags = {
+    Name = "cleanup-snapshots-${local.naming_suffix}"
+  }
+}
+
+resource "aws_iam_role" "cleanup_snapshots" {
+  name = "${var.pipeline_name}-${var.namespace}-cleanup-snapshots"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+
+  tags = {
+    Name = "cleanup-snapshots-${local.naming_suffix}"
```

Comment:
Given the Lambda is only cleaning up EC2 snapshots can the tag include `ec2` eg. `lambda-role-cleanup-ec2-snapshots` so that the actual type and what it does is captured in the tag.